### PR TITLE
Make cat/cat_out native function that rejects scalar inputs.

### PR DESCRIPTION
**Pull request by gchanan**

### CODE REVIEW

**Review by colesbury — COMMENTED** [review]
lgtm

File: aten/src/ATen/native/TensorShape.cpp [hunk 1]
```diff
@@ -8,6 +8,25 @@
 namespace at {
 namespace native {
 
+static void check_cat_no_zero_dim(TensorList tensors) {
+  for(size_t i = 0; i < tensors.size(); ++i) {
```
Comment (colesbury) [inline aten/src/ATen/native/TensorShape.cpp]: If you don't need `i` you can just use a for-each loop:

```
for (auto& t : tensors) {
  ...
}
```

Comment (gchanan) [reply aten/src/ATen/native/TensorShape.cpp]: oh, I was going to add a position in the error message, let me do that.

**Comment by gchanan** [comment]
onnx failure looks unrelated.

**Pull request merged**